exercises/practice/matrix/src/lib.rs

```diff
@@ -0,0 +1,17 @@
+pub struct Matrix {
+    // Implement your Matrix struct
+}
```

Comment:
Did you consider any other design than struct + methods? as far as the upstream test cases are concerned, it could just be two plain functions without an intermediate data structure. The way the tests are written now, this struct + methods approach cannot be avoided.

Reply:
I did not just because I was basing the design on the Python and JavaScript implementations (only other languages I'm comfortable with) and both implemented classes for the exercise. 

I will say that the tests on the other languages would agree with your thought process. None of them are really in need of having to keep the state of the input because it's always just checking either row or column but not both. The only thing I would say goes in favor of having a struct is if this were a real life program, I would want to use a struct so I can keep the state of the input and reuse it when needed.

I can re-write as two functions, though, if you think it would be more beneficial to the track overall.